tools/admin_ui_selenium_test.py

```
@@ -0,0 +1,393 @@
+        path = Path(explicit_path).expanduser()
+        if path.is_file():
+            return str(path)
+        fail(f"ChromeDriver not found at {path}")
+
+    path_candidate = shutil.which("chromedriver")
+    if path_candidate:
+        return path_candidate
+
+    cached = sorted(
+        Path.home().glob(".cache/selenium/chromedriver/*/*/chromedriver"),
+        key=lambda item: str(item),
```

Ruff (0.15.6)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)